data/scripts/bin/update-hosts.sh.template

```diff
@@ -49,13 +36,39 @@ host_env_base64=$(base64 -w 0 /etc/assisted/rendezvous-host.env)
 placeholder_base64=$(echo -n '{{.RendezvousHostEnvPlaceholder}}' | base64)
 ignition="${ignition//$placeholder_base64/$host_env_base64}"
 
-# Update installer-args and ignition for each host
-if [[ -n ${host_ids} ]]; then
+# Waiting for the cluster-id to be available
+until [[ -n ${cluster_id} ]]; do
+    echo "Querying assisted-service for cluster-id..."
+    cluster_id=$(curl_assisted_service "/clusters" GET | jq -r '.[0].id')
+    sleep 1
+done
+echo "Fetched cluster-id: $cluster_id"
+
+# Updating hosts can be done before starting the installation
+until [[ $cluster_status == "preparing-for-installation" ]]; do
```

Comment:
In the interactive workflow there could be a (very) tiny chance to miss updating an host before the installation is started by the UI - but to address that an explicit synchronization mechanism would be required. Anyhow the patch lgtm, since it's definitely an improvement in respect to the previous version.

Reply:
Right, I've was worried for exactly that when introducing this change:) But we indeed don't have such sync mechanism yet. So we could start with this for now.

Reply:
Have we considered building this functionality directly into assisted-service?
This does seem a bit race-y.